File: Week4/templates/B_fiftyshades.cpp
#include <bits/stdc++.h>
using namespace std;

bool fiftyshades(string& s){

    for(char& c : s){
        c = tolower(c); // std is helpful!
        // can also do:
        // if(c >= 'A' && 'c' <= 'Z') c += 'a' - 'A';
        // or something like that
    }
    if(s.find("rose") != string::npos) return true;
    if(s.find("pink") != string::npos) return true;
    // More useful stdlib :)

    // We could also do something like: 
    /*
    for(int i = 0; i + 4 <= s.size(); i++){
        if(s.substr(i, 4) == "rose") return true;
        if(s.substr(i, 4) == "pink") return true;
    }
    */
    return false;
}

int main(){
    int q;
    cin >> q;
    int total = 0;
    for(int i = 0; i < q; i++){
        string s;
        cin >> s;
        if(fiftyshades(s)) total++;
    }
    if(total == 0) cout << "I must watch Star Wars with my daughter" << endl;
    else cout << total << endl;
}

File: Week4/templates/B_fiftyshades.py


def fiftyshades(s):
    s = s.lower() # helpful to know library functions :)
    # We could also do something like:
    """
    for i in range(len(s)):
        if s[i] >= 'A' and s[i] <= 'Z':
            s[i] += 'a' - 'A'
    """
    # Check if s contains "pink" or "rose"
    if "pink" in s or "rose" in s:
        return True
    
    # More useful library functions :)

    # We could also do something like:
    """
    for i in range(max(len(s) - 3, 0)):
        if s[i:i+4] == "pink" or s[i:i+4] == "rose":
            return True
    """

    
    return False

q = int(input())
ttl = 0
for i in range(q):
    s = input()
    if fiftyshades(s):
        ttl += 1

if ttl == 0:
    print("I must watch Star Wars with my daughter")
else:
    print(ttl)





